File: PST/FUEL/src/FUEL/dlgTestComplete.vb
Imports DevComponents.DotNetBar
Imports DevComponents.DotNetBar.Controls
Imports DevComponents.DotNetBar.Metro
Imports FUEL.My.Resources
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Drawing
Imports System.Runtime.CompilerServices
Imports System.Windows.Forms

Namespace FUEL
    <DesignerGenerated> _
    Public Class dlgTestComplete
        Inherits MetroForm
        ' Methods
        Public Sub New(ByVal Passed As Boolean)
            AddHandler MyBase.Shown, New EventHandler(AddressOf Me.dlgTestComplete_Shown)
            dlgTestComplete.__ENCAddToList(Me)
            Me.InitializeComponent
            Me._Passed = Passed
        End Sub

        <DebuggerNonUserCode> _
        Private Shared Sub __ENCAddToList(ByVal value As Object)
            SyncLock dlgTestComplete.__ENCList
                If (dlgTestComplete.__ENCList.Count = dlgTestComplete.__ENCList.Capacity) Then
                    Dim index As Integer = 0
                    Dim num3 As Integer = (dlgTestComplete.__ENCList.Count - 1)
                    Dim num2 As Integer = 0
                    Do While True
                        Dim num4 As Integer = num3
                        If (num2 > num4) Then
                            dlgTestComplete.__ENCList.RemoveRange(index, (dlgTestComplete.__ENCList.Count - index))
                            dlgTestComplete.__ENCList.Capacity = dlgTestComplete.__ENCList.Count
                            Exit Do
                        End If
                        Dim reference As WeakReference = dlgTestComplete.__ENCList(num2)
                        If reference.IsAlive Then
                            If (num2 <> index) Then
                                dlgTestComplete.__ENCList(index) = dlgTestComplete.__ENCList(num2)
                            End If
                            index += 1
                        End If
                        num2 += 1
                    Loop
                End If
                dlgTestComplete.__ENCList.Add(New WeakReference(value))
            End SyncLock
        End Sub

        <DebuggerNonUserCode> _
        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            Try 
                If (If((Not disposing OrElse (Me.components Is Nothing)), 0, 1) <> 0) Then
                    Me.components.Dispose
                End If
            Finally
                MyBase.Dispose(disposing)
            End Try
        End Sub

        Private Sub dlgTestComplete_Shown(ByVal sender As Object, ByVal e As EventArgs)
            If Me._Passed Then
                Me.PictureBox1.Image = Resources.happy_sm
            Else
                Me.PictureBox1.Image = Resources.frown_sm
                Me.ReflectionLabel2.Text = Strings.Replace(Me.ReflectionLabel2.Text, "#20C500", "#B02B2C", 1, -1, CompareMethod.Binary)
                Me.ReflectionLabel2.Text = Strings.Replace(Me.ReflectionLabel2.Text, "Passed", "Failed", 1, -1, CompareMethod.Binary)
            End If
        End Sub

        <DebuggerStepThrough> _
        Private Sub InitializeComponent()
            Dim manager As New ComponentResourceManager(GetType(dlgTestComplete))
            Me.PictureBox1 = New PictureBox
            Me.ReflectionLabel1 = New ReflectionLabel
            Me.ReflectionLabel2 = New ReflectionLabel
            Me.cmdOkay = New ButtonX
            DirectCast(Me.PictureBox1, ISupportInitialize).BeginInit
            Me.SuspendLayout
            Me.PictureBox1.BackColor = Color.Transparent
            Me.PictureBox1.ForeColor = Color.Black
            Me.PictureBox1.Image = DirectCast(manager.GetObject("PictureBox1.Image"), Image)
            Dim point2 As New Point(9, 9)
            Me.PictureBox1.Location = point2
            Dim padding2 As New Padding(0)
            Me.PictureBox1.Margin = padding2
            Me.PictureBox1.Name = "PictureBox1"
            Dim size2 As New Size(&H80, &H7D)
            Me.PictureBox1.Size = size2
            Me.PictureBox1.SizeMode = PictureBoxSizeMode.StretchImage
            Me.PictureBox1.TabIndex = 2
            Me.PictureBox1.TabStop = False
            Me.ReflectionLabel1.BackgroundStyle.CornerType = eCornerType.Square
            point2 = New Point(140, 0)
            Me.ReflectionLabel1.Location = point2
            Me.ReflectionLabel1.Name = "ReflectionLabel1"
            size2 = New Size(290, 70)
            Me.ReflectionLabel1.Size = size2
            Me.ReflectionLabel1.TabIndex = 3
            Me.ReflectionLabel1.Text = "<b><font size=""+10""><i>Test Complete</i></font></b>"
            Me.ReflectionLabel2.BackgroundStyle.CornerType = eCornerType.Square
            point2 = New Point(140, &H40)
            Me.ReflectionLabel2.Location = point2
            Me.ReflectionLabel2.Name = "ReflectionLabel2"
            size2 = New Size(290, 70)
            Me.ReflectionLabel2.Size = size2
            Me.ReflectionLabel2.TabIndex = 4
            Me.ReflectionLabel2.Text = "<b><font size=""+12""><i>Status:  </i><font color=""#B02B2C""><font color=""#20C500"">Passed</font></font></font></b>"
            Me.cmdOkay.AccessibleRole = AccessibleRole.PushButton
            Me.cmdOkay.ColorTable = eButtonColor.OrangeWithBackground
            point2 = New Point(&HB1, 140)
            Me.cmdOkay.Location = point2
            Me.cmdOkay.Name = "cmdOkay"
            size2 = New Size(&H4B, &H17)
            Me.cmdOkay.Size = size2
            Me.cmdOkay.Style = eDotNetBarStyle.StyleManagerControlled
            Me.cmdOkay.TabIndex = 5
            Me.cmdOkay.Text = "Okay"
            Me.AcceptButton = Me.cmdOkay
            Dim ef2 As New SizeF(6!, 13!)
            Me.AutoScaleDimensions = ef2
            Me.AutoScaleMode = AutoScaleMode.Font
            size2 = New Size(&H1B3, &HA9)
            Me.ClientSize = size2
            Me.Controls.Add(Me.cmdOkay)
            Me.Controls.Add(Me.ReflectionLabel2)
            Me.Controls.Add(Me.ReflectionLabel1)
            Me.Controls.Add(Me.PictureBox1)
            Me.DoubleBuffered = True
            Me.FormBorderStyle = FormBorderStyle.FixedDialog
            Me.MaximizeBox = False
            Me.MinimizeBox = False
            Me.Name = "dlgTestComplete"
            Me.ShowInTaskbar = False
            Me.StartPosition = FormStartPosition.CenterParent
            Me.Text = "Test Complete"
            DirectCast(Me.PictureBox1, ISupportInitialize).EndInit
            Me.ResumeLayout(False)
        End Sub

        Private Sub OK_Button_Click(ByVal sender As Object, ByVal e As EventArgs)
            Me.DialogResult = DialogResult.OK
            Me.Close
        End Sub


        ' Properties
        Friend Overridable Property PictureBox1 As PictureBox
            <DebuggerNonUserCode> _
            Get
                Return Me._PictureBox1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As PictureBox)
                Me._PictureBox1 = WithEventsValue
            End Set
        End Property

        Friend Overridable Property ReflectionLabel1 As ReflectionLabel
            <DebuggerNonUserCode> _
            Get
                Return Me._ReflectionLabel1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As ReflectionLabel)
                Me._ReflectionLabel1 = WithEventsValue
            End Set
        End Property

        Friend Overridable Property ReflectionLabel2 As ReflectionLabel
            <DebuggerNonUserCode> _
            Get
                Return Me._ReflectionLabel2
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As ReflectionLabel)
                Me._ReflectionLabel2 = WithEventsValue
            End Set
        End Property

        Friend Overridable Property cmdOkay As ButtonX
            <DebuggerNonUserCode> _
            Get
                Return Me._cmdOkay
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As ButtonX)
                Dim handler As EventHandler = New EventHandler(AddressOf Me.OK_Button_Click)
                If Not Object.ReferenceEquals(Me._cmdOkay, Nothing) Then
                    RemoveHandler Me._cmdOkay.Click, handler
                End If
                Me._cmdOkay = WithEventsValue
                If Not Object.ReferenceEquals(Me._cmdOkay, Nothing) Then
                    AddHandler Me._cmdOkay.Click, handler
                End If
            End Set
        End Property


        ' Fields
        Private Shared __ENCList As List(Of WeakReference) = New List(Of WeakReference)
        Private components As IContainer
        <AccessedThroughProperty("PictureBox1")> _
        Private _PictureBox1 As PictureBox
        <AccessedThroughProperty("ReflectionLabel1")> _
        Private _ReflectionLabel1 As ReflectionLabel
        <AccessedThroughProperty("ReflectionLabel2")> _
        Private _ReflectionLabel2 As ReflectionLabel
        <AccessedThroughProperty("cmdOkay")> _
        Private _cmdOkay As ButtonX
        Private _Passed As Boolean
    End Class
End Namespace

File: PST/FUEL Source Code/FUEL/Backup/dlgTestComplete.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.dlgTestComplete
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using DevComponents.DotNetBar;
using DevComponents.DotNetBar.Controls;
using DevComponents.DotNetBar.Metro;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Diagnostics;
using System.Drawing;
using System.Runtime.CompilerServices;
using System.Windows.Forms;

namespace FUEL
{
  [DesignerGenerated]
  public class dlgTestComplete : MetroForm
  {
    private static List<WeakReference> __ENCList = new List<WeakReference>();
    private IContainer components;
    [AccessedThroughProperty("PictureBox1")]
    private PictureBox _PictureBox1;
    [AccessedThroughProperty("ReflectionLabel1")]
    private ReflectionLabel _ReflectionLabel1;
    [AccessedThroughProperty("ReflectionLabel2")]
    private ReflectionLabel _ReflectionLabel2;
    [AccessedThroughProperty("cmdOkay")]
    private ButtonX _cmdOkay;
    private bool _Passed;

    [DebuggerNonUserCode]
    static dlgTestComplete()
    {
    }

    [DebuggerNonUserCode]
    private static void __ENCAddToList(object value)
    {
      lock (dlgTestComplete.__ENCList)
      {
        if (dlgTestComplete.__ENCList.Count == dlgTestComplete.__ENCList.Capacity)
        {
          int index1 = 0;
          int num = checked (dlgTestComplete.__ENCList.Count - 1);
          int index2 = 0;
          while (index2 <= num)
          {
            if (dlgTestComplete.__ENCList[index2].IsAlive)
            {
              if (index2 != index1)
                dlgTestComplete.__ENCList[index1] = dlgTestComplete.__ENCList[index2];
              checked { ++index1; }
            }
            checked { ++index2; }
          }
          dlgTestComplete.__ENCList.RemoveRange(index1, checked (dlgTestComplete.__ENCList.Count - index1));
          dlgTestComplete.__ENCList.Capacity = dlgTestComplete.__ENCList.Count;
        }
        dlgTestComplete.__ENCList.Add(new WeakReference(RuntimeHelpers.GetObjectValue(value)));
      }
    }

    [DebuggerNonUserCode]
    protected override void Dispose(bool disposing)
    {
      try
      {
        if (!disposing || this.components == null)
          return;
        this.components.Dispose();
      }
      finally
      {
        base.Dispose(disposing);
      }
    }

    [DebuggerStepThrough]
    private void InitializeComponent()
    {
      ComponentResourceManager componentResourceManager = new ComponentResourceManager(typeof (dlgTestComplete));
      this.PictureBox1 = new PictureBox();
      this.ReflectionLabel1 = new ReflectionLabel();
      this.ReflectionLabel2 = new ReflectionLabel();
      this.cmdOkay = new ButtonX();
      ((ISupportInitialize) this.PictureBox1).BeginInit();
      this.SuspendLayout();
      this.PictureBox1.BackColor = Color.Transparent;
      this.PictureBox1.ForeColor = Color.Black;
      this.PictureBox1.Image = (Image) componentResourceManager.GetObject("PictureBox1.Image");
      PictureBox pictureBox1_1 = this.PictureBox1;
      Point point1 = new Point(9, 9);
      Point point2 = point1;
      pictureBox1_1.Location = point2;
      this.PictureBox1.Margin = new System.Windows.Forms.Padding(0);
      this.PictureBox1.Name = "PictureBox1";
      PictureBox pictureBox1_2 = this.PictureBox1;
      Size size1 = new Size(128, 125);
      Size size2 = size1;
      pictureBox1_2.Size = size2;
      this.PictureBox1.SizeMode = PictureBoxSizeMode.StretchImage;
      this.PictureBox1.TabIndex = 2;
      this.PictureBox1.TabStop = false;
      this.ReflectionLabel1.BackgroundStyle.CornerType = eCornerType.Square;
      ReflectionLabel reflectionLabel1_1 = this.ReflectionLabel1;
      point1 = new Point(140, 0);
      Point point3 = point1;
      reflectionLabel1_1.Location = point3;
      this.ReflectionLabel1.Name = "ReflectionLabel1";
      ReflectionLabel reflectionLabel1_2 = this.ReflectionLabel1;
      size1 = new Size(290, 70);
      Size size3 = size1;
      reflectionLabel1_2.Size = size3;
      this.ReflectionLabel1.TabIndex = 3;
      this.ReflectionLabel1.Text = "<b><font size=\"+10\"><i>Test Complete</i></font></b>";
      this.ReflectionLabel2.BackgroundStyle.CornerType = eCornerType.Square;
      ReflectionLabel reflectionLabel2_1 = this.ReflectionLabel2;
      point1 = new Point(140, 64);
      Point point4 = point1;
      reflectionLabel2_1.Location = point4;
      this.ReflectionLabel2.Name = "ReflectionLabel2";
      ReflectionLabel reflectionLabel2_2 = this.ReflectionLabel2;
      size1 = new Size(290, 70);
      Size size4 = size1;
      reflectionLabel2_2.Size = size4;
      this.ReflectionLabel2.TabIndex = 4;
      this.ReflectionLabel2.Text = "<b><font size=\"+12\"><i>Status:  </i><font color=\"#B02B2C\"><font color=\"#20C500\">Passed</font></font></font></b>";
      this.cmdOkay.AccessibleRole = AccessibleRole.PushButton;
      this.cmdOkay.ColorTable = eButtonColor.OrangeWithBackground;
      ButtonX cmdOkay1 = this.cmdOkay;
      point1 = new Point(177, 140);
      Point point5 = point1;
      cmdOkay1.Location = point5;
      this.cmdOkay.Name = "cmdOkay";
      ButtonX cmdOkay2 = this.cmdOkay;
      size1 = new Size(75, 23);
      Size size5 = size1;
      cmdOkay2.Size = size5;
      this.cmdOkay.Style = eDotNetBarStyle.StyleManagerControlled;
      this.cmdOkay.TabIndex = 5;
      this.cmdOkay.Text = "Okay";
      this.AcceptButton = (IButtonControl) this.cmdOkay;
      this.AutoScaleDimensions = new SizeF(6f, 13f);
      this.AutoScaleMode = AutoScaleMode.Font;
      size1 = new Size(435, 169);
      this.ClientSize = size1;
      this.Controls.Add((Control) this.cmdOkay);
      this.Controls.Add((Control) this.ReflectionLabel2);
      this.Controls.Add((Control) this.ReflectionLabel1);
      this.Controls.Add((Control) this.PictureBox1);
      this.DoubleBuffered = true;
      this.FormBorderStyle = FormBorderStyle.FixedDialog;
      this.MaximizeBox = false;
      this.MinimizeBox = false;
      this.Name = nameof (dlgTestComplete);
      this.ShowInTaskbar = false;
      this.StartPosition = FormStartPosition.CenterParent;
      this.Text = "Test Complete";
      ((ISupportInitialize) this.PictureBox1).EndInit();
      this.ResumeLayout(false);
    }

    internal virtual PictureBox PictureBox1
    {
      [DebuggerNonUserCode] get => this._PictureBox1;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._PictureBox1 = value;
    }

    internal virtual ReflectionLabel ReflectionLabel1
    {
      [DebuggerNonUserCode] get => this._ReflectionLabel1;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._ReflectionLabel1 = value;
    }

    internal virtual ReflectionLabel ReflectionLabel2
    {
      [DebuggerNonUserCode] get => this._ReflectionLabel2;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._ReflectionLabel2 = value;
    }

    internal virtual ButtonX cmdOkay
    {
      [DebuggerNonUserCode] get => this._cmdOkay;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set
      {
        EventHandler eventHandler = new EventHandler(this.OK_Button_Click);
        if (this._cmdOkay != null)
          this._cmdOkay.Click -= eventHandler;
        this._cmdOkay = value;
        if (this._cmdOkay == null)
          return;
        this._cmdOkay.Click += eventHandler;
      }
    }

    public dlgTestComplete(bool Passed)
    {
      this.Shown += new EventHandler(this.dlgTestComplete_Shown);
      dlgTestComplete.__ENCAddToList((object) this);
      this.InitializeComponent();
      this._Passed = Passed;
    }

    private void OK_Button_Click(object sender, EventArgs e)
    {
      this.DialogResult = DialogResult.OK;
      this.Close();
    }

    private void dlgTestComplete_Shown(object sender, EventArgs e)
    {
      if (!this._Passed)
      {
        this.PictureBox1.Image = (Image) FUEL.My.Resources.Resources.frown_sm;
        this.ReflectionLabel2.Text = Strings.Replace(this.ReflectionLabel2.Text, "#20C500", "#B02B2C");
        this.ReflectionLabel2.Text = Strings.Replace(this.ReflectionLabel2.Text, "Passed", "Failed");
      }
      else
        this.PictureBox1.Image = (Image) FUEL.My.Resources.Resources.happy_sm;
    }
  }
}
